docs/Antora/modules/howto_guides/pages/howto_configure_instance.adoc

```diff
@@ -0,0 +1,291 @@
+    daq::InstancePtr instance = instanceBuilder.build();
+
+    return 0;
+}
+----
+====
+As in previous examples, developers can implement their own solution for the scheduler and use it in the Instance Builder.
+[tabs]
+====
+Cpp::
++
+[source,cpp]
+----
+#include <opendaq/opendaq.h>
+#include <iostream>
+
+class CustomSchedulerImpl : public ImplementationOfWeak<IScheduler>
+{
+    CustomSchedulerImpl(size_t workers)
+    {
+        // initializing
+    }
+    // implementation
+} 
+
+int main(int argc, const char* argv[])
+{
+    daq::SchedulerPtr scheduler = daq::SchedulerPtr(CustomSchedulerImpl(2));
+    daq::InstanceBuilderPtr instanceBuilder = daq::InstanceBuilder().setScheduler(scheduler);
+    daq::InstancePtr instance = instanceBuilder.build();
+
+    return 0;
+}
+----
+====
+
+== Configure openDAQ(TM) Root Device
+If, after creating the Instance object, the developer has not set the root device, the Instance will use the Client device as the root. Using the Instance Builder, developers can discover and connect to available devices using the method `getAvailableDevices`. In the Instance Builder, developers can forcibly set the available device.
+[tabs]
+====
+Cpp::
++
+[source,cpp]
+----
+#include <opendaq/opendaq.h>
+#include <iostream>
+
+int main(int argc, const char* argv[])
+{
+    daq::InstanceBuilderPtr instanceBuilder = daq::InstanceBuilder().setSchedulerWorkerNum(2);
+    daq::InstancePtr instance = instanceBuilder.build();
+
+    return 0;
+}
+----
+====
+As in the previous examples, developers can implement their own scheduler solution and use it in Instance Builder. In that mode, developers can discover and connect to an available device using method `getAvailableDevices`. In Instance Builder, developers can force set available devices.
+[tabs]
+====
+Cpp::
++
+[source,cpp]
+----
+#include <opendaq/opendaq.h>
+#include <iostream>
+
+int main(int argc, const char* argv[])
+{
+    daq::DeviceInfoPtr defaultRootDeviceInfo = daq::DeviceInfo("daqref://device0");
+    daq::InstanceBuilderPtr instanceBuilder = daq::InstanceBuilder().setDefaultRootDeviceInfo(defaultRootDeviceInfo);
+    daq::InstancePtr instance = instanceBuilder.build();
+
+    assert(instance.getAvailableDevices()[0] == defaultRootDeviceInfo);
+
+    return 0;
+}
+----
+====
+Developers can skip discovering available devices if they know the device address. For this purpose, the builder has the method `setRootDevice` with a connection string, which replaces the client device as the root device with a custom device.
+[tabs]
+====
+Cpp::
++
+[source,cpp]
+----
+#include <opendaq/opendaq.h>
+#include <iostream>
+
+int main(int argc, const char* argv[])
+{
+    daq::InstanceBuilderPtr instanceBuilder = daq::InstanceBuilder().setRootDevice("daqref://device0");
+    daq::InstancePtr instance = instanceBuilder.build();
+
+    assert(instance.getAvailableDevices()[0] == instanceBuilder);
+
+    return 0;
+}
+----
```

Comment:
Connecting to a device is a separate how-to. This how-to should focus on creating and configuring the instance, not using it post-creation.